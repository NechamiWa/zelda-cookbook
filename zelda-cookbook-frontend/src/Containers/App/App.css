@import url("https://fonts.googleapis.com/css?family=PT+Serif:700");
* {
  box-sizing: border-box; }

.App {
  display: grid;
  grid-template-columns: 340px auto; }

.sidebar {
  background: linear-gradient(to bottom, #1c3744 25%, rgba(221, 208, 182, 0.75));
  min-height: 100vh;
  padding: 20px; }

#small-zelda-header {
  background-image: url("../../assets/small-zelda-header.png");
  background-size: contain;
  background-repeat: no-repeat;
  background-color: #1c3744;
  box-shadow: 0 0 20px 10px #1c3744;
  margin: auto;
  height: 194px;
  width: 253px; }

#search-container {
  align-items: stretch;
  display: grid;
  justify-content: space-around;
  grid-template-columns: 80% auto;
  grid-template-rows: 1fr; }

#dropDown {
  grid-row-start: 2;
  justify-self: stretch; }

body {
  background-image: url("../../assets/floral-old-paper-or-parchment.jpg");
  background-size: cover;
  background-repeat: no-repeat;
  height: 100vh;
  width: 100vw; }
