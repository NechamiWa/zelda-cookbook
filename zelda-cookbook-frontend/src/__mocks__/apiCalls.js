/* eslint-disable */
export const getIngredients = jest.fn()
  .mockImplementation(() => ([{
    id: 1,
    category: 'food',
    duration: '0:30',
    effect: null,
    hearts: '0.5',
    name: 'Apple',
    resale: '3',
    type: 'Fruit',
    created_at: '2018-04-07T02:42:46.591Z',
    updated_at: '2018-04-07T02:42:46.591Z'
  }]))

export const getRecipes = jest.fn()
  .mockImplementation(() => ([{
    id: 106,
    category: 'recipe',
    hearts: 'Â¾',
    name: 'Baked Apple',
    notes: 'Open flame.',
    resale: '0',
    type: 'Restore Hearts',
    duration: null,
    strength: null,
    ingredient1: 'Apple',
    ingredient2: null,
    ingredient3: null,
    ingredient4: null,
    ingredient5: null,
    ingredient1_id: 2,
    ingredient2_id: null,
    ingredient3_id: null,
    ingredient4_id: null,
    ingredient5_id: null,
    created_at: '2018-04-09T20:52:52.577Z',
    updated_at: '2018-04-09T20:52:52.577Z'
  }]))