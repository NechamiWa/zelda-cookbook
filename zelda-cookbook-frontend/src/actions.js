// Recipes =================

export const populateRecipes = recipes => ({
  type: 'POPULATE_RECIPES',
  recipes
});

// Ingredients =============

export const populateIngredients = ingredients => ({
  type: 'POPULATE_INGREDIENTS',
  ingredients
});

// Filter ==================

export const addIngredientFilter = ingredient => ({
  type: 'ADD_INGREDIENT_FILTER',
  ingredient
});

export const removeIngredientFilter = id => ({
  type: 'REMOVE_INGREDIENT_FILTER',
  id
});

export const updateNameFilter = name => ({
  type: 'UPDATE_NAME_FILTER',
  name
});

export const addTypeFilter = recipeType => ({
  type: 'ADD_TYPE_FILTER',
  recipeType
});

export const removeTypeFilter = () => ({
  type: 'REMOVE_TYPE_FILTER'
});