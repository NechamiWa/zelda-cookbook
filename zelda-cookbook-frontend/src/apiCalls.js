export const getIngredients = async () => {
  try {
     const initialFetch = await fetch('http://localhost:8000/api/v1/ingredients');
    //const initialFetch = await fetch('https://zelda-cookbook-backend.herokuapp.com/api/v1/ingredients');
    return await initialFetch.json();
  } catch(error) {
    throw Error('Failed to get ingredients from database')
  }
}

export const getRecipes = async () => {
  try {
     const initialFetch = await fetch('http://localhost:8000/api/v1/recipes');
    //const initialFetch = await fetch('https://zelda-cookbook-backend.herokuapp.com/api/v1/recipes');
    return await initialFetch.json();
  } catch(error) {
    throw Error('Failed to get recipes from database')
  }
}