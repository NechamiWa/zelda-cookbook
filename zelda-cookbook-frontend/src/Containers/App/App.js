import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getRecipes, getIngredients } from '../../apiCalls';
import Filter from '../../Components/Filter/Filter';
import RecipeContainer from '../RecipeContainer/RecipeContainer';
import { populateRecipes, populateIngredients } from '../../actions';
import './App.css';
// import { Route } from 'react-router-dom';

export class App extends Component {
  
  componentDidMount = async () => {
    try {
      const ingredients = await getIngredients();
      const recipes = await getRecipes();
      this.props.populateRecipes(recipes);
      this.props.populateIngredients(ingredients);
    } catch (error) {
      throw Error(error);
    }
  }

  render(){
    return (
      <div className="App">
        <aside className="sidebar">
          <section id="small-zelda-header"></section>
          <Filter />
        </aside>
        <RecipeContainer />
      </div>
    );
  }
}

const { func } = PropTypes;

App.propTypes = {
  populateRecipes: func,
  populateIngredients: func
};

export const MDTP = dispatch => ({
  populateRecipes: recipes => dispatch(populateRecipes(recipes)),
  populateIngredients: ingredients => dispatch(populateIngredients(ingredients))
});

export default connect(null, MDTP)(App);
