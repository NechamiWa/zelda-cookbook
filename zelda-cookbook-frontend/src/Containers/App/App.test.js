/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { App, MDTP } from './App';
import * as apiCalls from '../../apiCalls';
import { populateRecipes, populateIngredients } from '../../actions';

jest.mock('../../apiCalls')

describe('App', () => {
  let wrapper;
  let mockPopulateIngredients = jest.fn();
  let mockPopulateRecipes = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<App
      populateRecipes = { mockPopulateRecipes }
      populateIngredients = { mockPopulateIngredients }
    />);
    
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('should call getIngredients', () => {
      expect(apiCalls.getIngredients).toHaveBeenCalled();
    });

    it('should call getRecipes', () => {
      expect(apiCalls.getRecipes).toHaveBeenCalled();
    });

    it('should call populateRecipes with a recipes array', () => {
      const expected = [{
        id: 106,
        category: 'recipe',
        hearts: '¾',
        name: 'Baked Apple',
        notes: 'Open flame.',
        resale: '0',
        type: 'Restore Hearts',
        duration: null,
        strength: null,
        ingredient1: 'Apple',
        ingredient2: null,
        ingredient3: null,
        ingredient4: null,
        ingredient5: null,
        ingredient1_id: 2,
        ingredient2_id: null,
        ingredient3_id: null,
        ingredient4_id: null,
        ingredient5_id: null,
        created_at: '2018-04-09T20:52:52.577Z',
        updated_at: '2018-04-09T20:52:52.577Z'
      }]

      expect(mockPopulateRecipes).toHaveBeenCalledWith(expected)
    });

    it('should call populateIngredients with an ingredients array', () => {
      const expected = [{
        id: 1,
        category: 'food',
        duration: '0:30',
        effect: null,
        hearts: '0.5',
        name: 'Apple',
        resale: '3',
        type: 'Fruit',
        created_at: '2018-04-07T02:42:46.591Z',
        updated_at: '2018-04-07T02:42:46.591Z'
      }]
      expect(mockPopulateIngredients).toHaveBeenCalledWith(expected);
    })
  });

  describe('MDTP', () => {
    it('should call dispatch when populateRecipes and populateIngredients are called', () => {
      const mockDispatch = jest.fn();
      const mockIngredients = [{
        id: 1,
        category: 'food',
        duration: '0:30',
        effect: null,
        hearts: '0.5',
        name: 'Apple',
        resale: '3',
        type: 'Fruit',
        created_at: '2018-04-07T02:42:46.591Z',
        updated_at: '2018-04-07T02:42:46.591Z'
      }];
      const mockRecipes = [{
        id: 106,
        category: 'recipe',
        hearts: '¾',
        name: 'Baked Apple',
        notes: 'Open flame.',
        resale: '0',
        type: 'Restore Hearts',
        duration: null,
        strength: null,
        ingredient1: 2,
        ingredient2: null,
        ingredient3: null,
        ingredient4: null,
        ingredient5: null,
        created_at: '2018-04-09T20:52:52.577Z',
        updated_at: '2018-04-09T20:52:52.577Z'
      }];
      const result = MDTP(mockDispatch);

      result.populateRecipes(mockRecipes);
      expect(mockDispatch).toHaveBeenCalledWith(populateRecipes(mockRecipes));
      result.populateIngredients(mockIngredients);
      expect(mockDispatch).toHaveBeenCalledWith(populateIngredients(mockIngredients));
    });
  })
});
