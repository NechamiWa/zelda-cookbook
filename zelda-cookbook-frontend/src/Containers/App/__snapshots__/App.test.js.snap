// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <aside
    className="sidebar"
  >
    <section
      id="small-zelda-header"
    />
    <Filter />
  </aside>
  <Connect(RecipeContainer) />
</div>
`;
